src/fiatconnect/saga.ts

```diff
@@ -427,33 +439,131 @@ export function* _getQuotes({
+    const normalizedQuote = _selectQuoteMatchingFiatAccount({
+      normalizedQuotes,
+      fiatAccount,
+    })
+    if (normalizedQuote) return { normalizedQuote, selectedFiatAccount: fiatAccount }
+  }
+
+  throw new Error('Could not find a fiat account matching any provided quote')
+}
+
+/**
+ * Given some quote parameters, fetches quotes from a single provider and returns the quote to use
+ * as well as a suitable fiat account that the user has on file that can be used for the quote.
+ *
+ * If a fiat account is provided to this function, a quote will be selected from the provider's list
+ * of returned quotes which is eligible to be used with the given fiat account. If no quote is found
+ * that matches the given fiat account, throws an error.
+ *
+ * If no fiat account is provided, both quotes and on-file fiat accounts will be fetched from
+ * the chosen provider, and will be tested against eachother to find a match. If multiple quote-account
+ * matches exist, one will be chosen at random. If no matching pair exists, throws an error.
+ **/
 export function* _getSpecificQuote({
   digitalAsset,
   cryptoAmount,
   flow,
   providerId,
-  fiatAccountType,
+  fiatAccount,
 }: {
   digitalAsset: CiCoCurrency
   cryptoAmount: number
   flow: CICOFlow
   providerId: string
-  fiatAccountType: FiatAccountType
+  fiatAccount?: FiatAccount
 }) {
+  // Despite fetching quotes for a single provider, there still may be multiple quotes, since a quote
+  // object is generated for each Fiat Account Schema supported by the provider.
   const quotes: (FiatConnectQuoteSuccess | FiatConnectQuoteError)[] = yield call(_getQuotes, {
     flow,
     digitalAsset,
     cryptoAmount,
     providerIds: [providerId],
   })
   const normalizedQuotes = normalizeFiatConnectQuotes(flow, quotes)
-  const normalizedQuote = normalizedQuotes.find((q) => q.getFiatAccountType() === fiatAccountType)
+
+  // If no account was provided, we need to fetch accounts from the provider and find a matching quote-account pair
+  if (!fiatAccount) {
+    return (yield call(_selectQuoteAndFiatAccount, {
+      normalizedQuotes,
+      providerId,
+    })) as {
+      normalizedQuote: FiatConnectQuote
+      selectedFiatAccount: FiatAccount
+    }
+  }
+  // Otherwise, just find a quote that matches the selected account
+  const normalizedQuote = _selectQuoteMatchingFiatAccount({
+    normalizedQuotes,
+    fiatAccount,
+  })
   if (!normalizedQuote) {
     throw new Error('Could not find quote')
   }
-  return normalizedQuote
+  return {
+    normalizedQuote,
+    selectedFiatAccount: fiatAccount,
+  }
 }
 
+/**
+ * Fetches a fiat account from a given provider according to a set of optional filters. Possible filters are
+ * fiatAccountId, fiatAccountType, and fiatAccountSchema. Multiple filters may be used in conjunction. If, after
+ * filtering, multiple possible accounts remain, a random one matching the given criteria is returned. Returns null
```

Comment:
```suggestion
 * filtering, multiple possible accounts remain, an arbitrary one matching the given criteria is returned. Returns null
```
it's not random, it's arbitrary

Reply:
colloquially random and arbitrary are used the same way, but formally, they are quite different